### PR TITLE
first pass at support for embedding within a page

### PR DESCRIPTION
**Pull request by paulfitz**
Until now, Grist takes over the entire page it is placed within. With this change, it can be more polite and take over a particular element.

Moving within pages within Grist fails (at least on Firefox), I have a small tweak in hand to grist-core to make it work (https://github.com/gristlabs/grist-core/pull/596).

### CODE REVIEW

**Comment by alexmojaki** [comment]
This looks cool and interesting, but it looks like it might not work if `initialData` is a URL that requires authenticated cookies to download, based on:

- https://stackoverflow.com/questions/65336664/iframe-content-does-not-use-parent-cookies-when-using-srcdoc-in-firefox-works-i
- https://bugzilla.mozilla.org/show_bug.cgi?id=1741489

Can you investigate?

**Comment by paulfitz** [comment]
That's a good point @alexmojaki. I added that combination as something not supported yet in the README. Since it is an accepted defect in firefox, there's some chance it will get fixed, although priority set for it is not high. For the narrow problem of loading the csv, we could make a workaround, where we do it on the main page and pass the data into the iframe. Kind of a drag, but wouldn't change the structure of solution. I'd worry there might be other problems; e.g. I'd hoped custom widgets could be served side by side with doc (just as an example).

**Comment by paulfitz** [comment]
I updated the text since the problem wouldn't be just for `csv`s. I suggested a workaround, but should check definitively that it works.

File: README.md [hunk 1]
```diff
@@ -58,6 +58,8 @@ Browsers don't have native support for Grist [yet :-)] but you can make a little
   * After that, it is just a case of putting a `.grist` file on your server beside this `.html` file, and filing in the options to `bootstrapGrist`.
   * You can also pass `initialData: 'path/to/data.csv'` to import a CSV file into a new table. In this case `initialFile` is optional.
   * You can also pass `elementId: 'element-id` to open Grist within an element in your page.
+    - If using `elementId`, and serving a `.grist` or `.csv` file at a URL that requires cookie-based authentication, be aware that Firefox may not support this yet ([bug report](https://bugzilla.mozilla.org/show_bug.cgi?id=1741489)).
+	- In that case, you can include a small wrapper page for your document as above, and embed it as an iframe yourself.
```
Comment (alexmojaki) [inline README.md:62]: So in general will cookies from the parent page be included from within an iframe with the same origin?

**Review by alexmojaki — APPROVED** [review]
Cool!

File: README.md (hunk 1 above)
Comment (paulfitz) [reply README.md:62]: I think this is the case? In an experiment with `SameSite: Strict` (for which I can confirm seeing the no-cookie problem for a `srcdoc` iframe in Firefox but not Chrome) I see cookies in a regular iframe. This is for pages on the same domain, with `path` set to a common stem, and no extra settings on the iframe (such as sandboxing).

**Pull request merged**